doc/userdoc/styleguide/styleguide.rst

```diff
@@ -0,0 +1,566 @@
+
+       for i in line:
+           x = i
+           print(thing.x)
+
+For other code styles use the code-block directive (you can also use this one
+for python)
+
+
+
+\   .. code-block:: cpp
+
+Rendered as
+
+    .. code-block:: cpp
+
+       int main()
+       {
+         cout << "Hello, World!";
+         return 0;
+       }
+
+
+For input/output examples use the following syntax
+
+::
+
+   >>> input begins with 3, right-angled brackets
+   output is directly below input without any brackets
+
+
+Directives
+~~~~~~~~~~~
+
+This is not a complete guide to the directives of reStructuredText. For more
+options see `Sphinx directives <https://www.sphinx-doc.org/en/master/usage/
+restructuredtext/directives.html>`_ or `Docutils section on directives
+<http://docutils.sourceforge.net/docs/ref/rst/directives.html>`_.
+
+Equations
+^^^^^^^^^
+
+For equations, use LaTeX markup:
+
+::
+
+    .. math::
+
+            f(x) = \int_{-\infty}^{\infty} \hat f(\xi)\ e^{2 \pi i x \xi}\,d\xi,
+
+Rendered as
+
+
+    .. math::
+
+            f(x) = \int_{-\infty}^{\infty} \hat f(\xi)\ e^{2 \pi i x \xi}\,d\xi,
+
+
+Admonitions
+^^^^^^^^^^^
+
+Use admonitions to draw readers attention to a particular point. Possible
+admonitions types include
+
+"see also", "attention", "caution", "danger", "error", "hint", "important",
+"note", "tip", "warning", "admonition"
+
+If you want a custom admonition use
+
+::
+
+   .. admonition:: custom name
+
+         Here is some text
+
+Rendered as
+
+
+   .. admonition:: custom name
+
+         Here is some text
+
+
+Roles
+~~~~~
+
+See `the roles section in the Sphinx docs <https://www.sphinx-doc.org/en/
+master/usage/restructuredtext/roles.html>`_ for details.
+
+In-text math
+^^^^^^^^^^^^
+
+For in-text math, use
+
+::
+
+   Now we can see :math:`x=1` for this example.
+
+This will be rendered as
+
+
+   Now we can see :math:`x=1` for this example.
+
+Referencing
+^^^^^^^^^^^
+
+For referencing reStructuredText files in repository, use
+
+::
+
+   :doc:`file`
+
+::
+
+   :doc:`custom name <path/file>`
+
+This will be rendered as
+
+   :doc:`file`
+
+   :doc:`custom name <file>`
+
+
+For section headings, you can use either of the two examples below. But to
+reference figures or arbitrary places in a file, you must include a custom
+name (second example) in the reference for it to work.
+
+::
+
+    :ref:`ref_name`
+
+or
+
+::
+
+    :ref:`custom name <ref_name>`
+
+Rendered as
+
+
+    :ref:`ref_name`
+
+    :ref:`custom name <ref_name>`
+
+
+.. note::
+
+    The `ref_name` needs to be above the section you want to reference with the
+    following syntax
+
+    ::
+
+        .. _ref_name:
+
+        secton_header
+        --------------
+
+Special links
+^^^^^^^^^^^^^
+
+.. attention::
+
+  The items in this section are still in development and have not been incorporated into nest:master!
+
+
+To link PyNEST API functions used in-text to the API reference page use the following syntax:
+
+::
+
+   :py:func:`.Create`
```

Comment:
Just out of interest: do you have plans to add these automatically by scanning all double-backtick-delimited code and matching against the list of known things (PyNEST functions, models, ...)? I think this would be really awesome and make the process of documentation writing much more pleasant.

Reply:
Indeed. 

Should this then result in a general recommendation that wherever a part (module, class, method, function) from the PyNEST API is mentioned in the documentation, one should use 
```
:py:func:`.Create`
```
instead of 
```
``Create``
```
And how would that then integrate with `nest.Create("iaf_psc_alpha")`?

Reply:
@jougs @heplesser The PR to add links is in draft mode  #2010.